test/unit/cinstance_test.rb

```diff
@@ -1038,4 +1038,28 @@ def create_pending_cinstance
     assert_not_includes ids_cinstances_like_list, id_cinstance_with_another_plan
     assert_not_includes ids_cinstances_like_list, id_service_contract_with_enterprise_plan
   end
+
+  test 'App ID can include special characters as defined in the RFC 6749' do
+    generate_key_rfc_6749 = -> { ("\x20".."\x7e").to_a.shuffle.slice(0, 254).join }
```

Comment:
```suggestion
    generate_key_rfc_6749 = -> { 255.times.map { ("\x20".."\x7e").sample }.join }
```

Reply:
@hallelujah It isn't really the same 🤔 In the test I've done, it tested absolutely all the characters between `("\x20".."\x7e")` (because there are less than 100), and the only random thing is the order, but all of them appear 😄 And this test is not to test the validation of the length, that's the purpose of the next test in the file, but the purpose of this one is to test that these characters are all accepted in the validation.

Reply:
OK but I did not understand the slice if you are sure that they are less than 255 chars.

Reply:
Right, I put it there while testing and forgot about it 👍 I'll remove it 😄 👍